Update version of MongoDB used for testing to 7.0

gsmet opened the pull request:
It is not the latest (8.0 was released in November) but it's probably the most common used and also it's consistent with the version used as a container.

Hopefully, this will be compatible with Ubuntu 22 and 24.

gsmet committed:
Update version of MongoDB used for testing to 7.0

It is not the latest (8.0 was released in November) but it's probably
the most common used and also it's consistent with the version used as a
container.

Hopefully, this will be compatible with Ubuntu 22 and 24.

gsmet committed:
Raise stop timeout hoping it will solve the Windows issue...

gsmet commented:
@loicmathieu this will be needed for the upcoming switch to Ubuntu 24. Unfortunately, things are not correctly stopped on Windows (or some other issues) which lead to CI stalling and being cancelled after 6 hours.

I think we are not the only one with the issue: https://github.com/flapdoodle-oss/de.flapdoodle.embed.mongo.spring/issues/47 .

I'm not entirely sure why MongoDB 7 won't stop properly on Windows... but it seems something is off.

loicmathieu commented:
Flapdoodle on Windows has always been a pain, I remember having the same issue when update to a major version of Flapdoodle.
I wonder if we should move more tests to containers and only keep the test resource that uses Flapdoodl for cluster tests (with replicas).
As a workaround we may disable those tests on windows and create a followup issue to re-enable them later (we already did that for some months).

gsmet committed:
Disable tests using replicas on Windows

It seems that the replicas are not correctly shut down and it leads to
job timeouts.

gsmet commented:
@gastaldi @geoand FYI probably something to merge early next year as CI will fail as soon as Ubuntu 24 becomes the default for our CI (it's already the case in my fork).
